_meetups/_meetups.json

```diff
@@ -0,0 +1,3 @@
+{
+  "permalink": "meetup/{{ page.date | date: '%Y/%m/%d' }}/"
```

Comment:
are we changing the permalink for a bunch of meetups? that seems wrong.

Reply:
So, <wince>I already started moving things around with previous commits</wince>

> where do we tell github "this is an eleventy build"?

I have an issue to implement next: #113

Reply:
> **TL;DR:** I believe that changing URI patterns is useful/needed and can/will set up 301 redirects to resolve old links to new links.

So yeah. I know that [cool URIs don't change](https://www.w3.org/Provider/Style/URI) and usually lead the fight to maintain that standard.

However, by the time a large architectural shift comes around, we usually have a better idea for how the content is being created, maintained, and used. There's no better time to apply lessons learned to new patterns than when you're changing engines and applying a fresh coat of paint.

After taking a long, hard look at the post content, a few patterns emerged:

1. there was confusion over the post creation date vs. the event date (see [README](https://github.com/austinjavascript/austinjavascript.github.io/tree/v2-eol#adding-a-new-post);
2. there was also complexities surrounding the use of the `when` field in post front matter (see [README](https://github.com/austinjavascript/austinjavascript.github.io/tree/v2-eol#adding-a-new-post) again); I saw evidence of the confusion after seeing several corrections in the commits;
3. finding post files could be an issue if you were expecting to find it by event date; post dates were often off by days and even weeks from event dates;
4. "meetup" posts have different content/purpose than "blog" posts and could/should have a different template; "meetup" posts were more akin to calendar events whereas "blog" posts contained a wider variety of notifications, reviews, and party announcements;

I attempted to address the points above by making the following changes:

1. file date == event date; (if file creation date is needed, look at `fs.stats.birthtime` or add a front matter field for that -- it is secondary meta info, after all)
2. assume all meetings start at 7:30pm and end at 9pm; in looking thru all the meetups, I found no major variation from those hours; the `when` front matter field was used for the event date (solved above) and the event time, which is now unnecessary;
3. since event dates are on the file name, locating past meetups is easy;
4. create separate source directories and use specialized templates for "meetup" and "blog" posts; also create separate navigation for the two (/meetup/2020/05/06/ and /post/2020/05/06/); this will allow us to have meetup and blog posts highlighted on the homepage in the future if so desired;

I believe that these changes will make creating and editing posts of all sorts a lot easier. I also think it will make the platform a lot more flexible going forward. I jumped a bit quickly perhaps and would love to slow down and discuss the changes more.

Also, I added a project card to do 301 redirects from old links to new. I hope that will alleviate all concerns about cool URIs.

Reply:
...an additional thought:

As I was normalizing the URLs and moving stuff around, I was expecting the need for redirects -- been thru this exercise more than once. Since we don't have access to the web server 301 redirects, we get to use meta refreshes. Keeping those files in a disposable directory (the old `posts`) would be convenient. Then all new content would go into `/meetup/` and `/post/` directories. Then, if we wish a year or two down the road, we could discard the `posts` redirects folder without affecting the others.

As I am writing this, I am thinking that there might be a way to intermingle the refresh pages with the "good" pages, but I'd have to look into it.

I also need to check-in on the real concern before I go off solving the wrong issue. Is it:

1. Not wanting orphaned links? (i.e., not maintaining [cool URIs](https://www.w3.org/Provider/Style/URI)
2. Not liking the separation of posts into `/meeting/` and `/post/` (or the naming)?
3. Both

Reply:
Here's the updated approach which should set a good direction for the future without breaking too much of the past:

* The `/posts/meetups/YYYY/MM/DD/` format for meetup permalinks will be preserved (which reduces the need to redirect by ~20). These meetups were also more recent and are still receiving a decent amount of traffic. 
* The meetup permalink date will be the event date [(as proposed in item number one above)](https://github.com/austinjavascript/austinjavascript.github.io/pull/112#discussion_r421031315). This requires ~20 redirects.
* All the WordPress (pre-Jekyll) meetup/post permalinks were moved off the root and converted to the new permalink style where possible. This requires ~60 redirects.  FWIW, most of these pages have averaged less than 10 pageviews per year and are decidedly on the long tail.
    * Meetups get the format mentioned above. 
    * "blog" posts get a `/posts/blog-slug-name/` treatment. This gets them off the site root to prevent namespace collisions (should someone post something such as `about` or `code-of-conduct`). There's more room for discussion here as there is still a possibility of naming collisions within the posts dir. It would be nice to have a `/posts/YYYY/MM/DD/blog-slug-name/` format but that would muck up the `/posts/meetups/YYYY/MM/DD/` format.
* The archive page was broken into more RESTful `/posts/` and `/posts/meetups/` pages.
* All redirects are mapped via a `redirects.json` file into a template that automatically generates the needed meta refresh pages.